web_app/src/components/auth/use-auth.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {createContext, useContext, useEffect, useRef, useState} from 'react'
+import firebase from '../../firebase-config'
+
+firebase.auth().useDeviceLanguage()
+
+type Auth = {
+  signIn: () => void
+  signOut: () => Promise<void>
+  user: firebase.User | null
+  admin: boolean | null
+}
+
+const authContext = createContext<Auth | null>(null);
+
+export const useAuth = () => {
+  return (
+    useContext(authContext) || {
+      signIn: () => undefined,
+      signOut: () => undefined,
+      user: firebase.auth().currentUser,
+      admin: null,
+    }
+  )
+}
+
+function useProvideAuth(): Auth {
+  const [user, setUser] = useState<firebase.User | null>(null)
+  const [admin, setAdmin] = useState<boolean | null>(null)
+
+  const signIn = () => {
+    firebase
+      .auth()
+      .signInWithRedirect(new firebase.auth.GoogleAuthProvider())
+      .then(()=> {
+        console.log('signed in')
+        }
+      )
+  }
+
+  const signOut = () => {
+    return firebase
+      .auth()
+      .signOut()
+      .then(() => {
+        console.log('signed out')
+        setUser(null)
+      })
+  }
+
+  const metadataUnsubscribe = useRef<(() => void) | null>(null);
```

Comment:
Denne og bruken av den kan fjernes. Dette henger igjen fra løsningen jeg brukte med functions og firestore for å sette claims på nye brukere.

Reply:
Ja, det trenger vi nok ikke, foreløpig